Assets/Global/Scripts/Configs/WeightedSelectionUtility.cs

```diff
@@ -0,0 +1,110 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine; 
+[System.Serializable]
+// For instantiating the item
+public class WeightableEntry<T>
+{ 
+    
+    public string Name;
+    public float Weight;
+    public T Entry;
+
+    public WeightableEntry(string name, float weight, T entry)
+    {
+        Name = name;
+        Weight = weight;
+        Entry = entry;
+    }
+}
+
+
+public static class WeightedSelectionUtility
+{
+    // Get a single random entry based on weight
+    public static WeightableEntry<T> GetRandomEntry<T>(List<WeightableEntry<T>> entries)
+    {
+        if (entries == null || entries.Count == 0)
+        {
+            Debug.LogWarning("[WARN] No entries available. [WeightedSelectionUtility.GetRandomEntry]");
+            return null;
+        }
+
+        float totalWeight = entries.Sum(entry => entry.Weight);
+
+        if (totalWeight <= 0)
+        {
+            Debug.LogWarning("[WARN] Total weight is zero or less. [WeightedSelectionUtility.GetRandomEntry]");
+            return null;
+        }
+
+        float randomValue = Random.Range(0, totalWeight);
+        float cumulativeWeight = 0;
+
+        foreach (var entry in entries)
+        {
+            cumulativeWeight += entry.Weight;
+            if (randomValue <= cumulativeWeight)
+            {
+                return entry;
+            }
+        }
+
+        Debug.LogWarning("[WARN] No valid entry selected. This should not happen. [WeightedSelectionUtility.GetRandomEntry]");
+        return null;
+    }
+
+    // Get multiple random entries, may allow duplicates
+    public static List<WeightableEntry<T>> GetMultipleRandomEntries<T>(List<WeightableEntry<T>> entries, int count, bool allowDuplicates = false)
+    {
+        if (count <= 0)
+        {
+            Debug.LogWarning("[WARN] Count must be greater than zero. [WeightedSelectionUtility.GetMultipleRandomEntries]");
+            return new List<WeightableEntry<T>>();
+        }
+
+        if (!allowDuplicates && entries.Count < count)
```

Comment:
if allowDuplicate is set to true and entries.Count < count returns true, it will not display the warning since false && true will always return false

Reply:
Yeah what Pim says, it's a warning. if 2 rewards r selected and there is one thing in the loottable and we DONT want duplicates, we warn. It's arguably a bad warning though, bit of an overkill I'd say cuz I think it'd be fine to just return one thing in that case lol